src/main/java/org/example/springintro/controller/BookController.java

```diff
@@ -28,8 +29,8 @@ public class BookController {
 
     @GetMapping
 
-    public List<BookDto> getAll() {
-        return bookService.findAll();
+    public List<BookDto> getAll(Pageable pageable) {
+        return bookService.findAll(pageable);
```

Comment:
Add `@ParameterObject @PageableDefault Pageable pageable` to search endpoint also